src/core/server/saved_objects/types.ts

```diff
@@ -291,6 +290,14 @@ export interface SavedObjectsType<Attributes = any> {
    * An optional map of {@link SavedObjectMigrationFn | migrations} or a function returning a map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
    */
   migrations?: SavedObjectMigrationMap | (() => SavedObjectMigrationMap);
+  /**
+   * An optional schema that can be used to validate the attributes of the type.
+   *
+   * When provided, calls to {@link SavedObjectsClient.create | create} will be validated against this schema.
+   *
+   * See {@link SavedObjectsValidationMap} for more details.
+   */
+  schemas?: SavedObjectsValidationMap<Attributes> | (() => SavedObjectsValidationMap<Attributes>);
```

Comment:
nit/comment: Using the plural of `schema` could be misleading in that it implies more than one schema. It's subtle but if we're restricting a saved object type to only have one `shape` per version, I'd prefer it if we went with singular `schema` here. OTOH, we use `migrations` and not `migration` so we might want to keep consistent.

Reply:
We usually use the plural for the property name when the property value is a map/record (or a function returning such types), so I'd say `schemas` is appropriate (and yea, also it's consistent with `migrations`)

The think bothering me here is the generic `Attributes` usage in the schemas. It's not quite accurate, because if a schema per version is registered, each schema / validation function will be against potentially different shapes. 

```ts
const myType: SavedObjectsType<MyLatestVersionAttributes> = {
  // ....
  schemas: {
   'lastVersion': someValidationFunctionUsingMyLatestVersionAttributes,
   'oldVersion': someValidationFunctionUsingOlderAttributes,
  }
}
```


But I don't see how we can improve this if we don't want to just use `unknown` or `any` here. (still, may it be preferable in term of design, idk)

Reply:
Yea, going back to `unknown` seems less dangerous and/or confusing (and, also, honest) for API consumers.